jpa/subsystem/src/main/java/org/jboss/as/jpa/classloader/JPADelegatingClassFileTransformer.java

```diff
@@ -44,13 +46,47 @@ public JPADelegatingClassFileTransformer(PersistenceUnitMetadata pu) {
     public byte[] transform(ClassLoader classLoader, String className, Class<?> aClass, ProtectionDomain protectionDomain, byte[] originalBuffer) throws
         IllegalClassFormatException {
         byte[] transformedBuffer = originalBuffer;
+        boolean transformed = false;
+
         for (javax.persistence.spi.ClassTransformer transformer : persistenceUnitMetadata.getTransformers()) {
             byte[] result = transformer.transform(classLoader, className, aClass, protectionDomain, transformedBuffer);
             if (result != null) {
                 transformedBuffer = result;
+                transformed = true;
+            }
+        }
+        return transformed ? transformedBuffer : null;
+    }
+
+    @Override
+    public ByteBuffer transform(ClassLoader classLoader, String className, ProtectionDomain protectionDomain, ByteBuffer classBytes)
+            throws IllegalArgumentException {
+        final int position = classBytes.position();
+        final int limit = classBytes.limit();
+        final byte[] bytes;
+        final Class<?> aClass = null;
+        if (classBytes.hasArray() && classBytes.arrayOffset() == 0 && position == 0 && limit == classBytes.capacity()) {
+            bytes = classBytes.array();
+        } else {
+            bytes = new byte[limit - position];
+            classBytes.get(bytes);
+            classBytes.position(position);
+        }
+        boolean transformed = false;
+        byte[] transformedBuffer = bytes;
+        for (javax.persistence.spi.ClassTransformer transformer : persistenceUnitMetadata.getTransformers()) {
+            byte[] result = new byte[0];
```

Comment:
This assignment and allocation is pointless.  If the transform call fails, an exception will be thrown; otherwise it will definitely assign `result`.

Reply:
removed.